src/client/dfs/duns.c

```diff
@@ -1304,11 +1307,33 @@ duns_link_cont(daos_handle_t poh, const char *cont, const char *path)
 		}
 		D_GOTO(err_link, rc);
 	}
+	if (backend_dfuse) {
+		struct stat finfo;
+		/*
+		 * This next stat will cause dfuse to lookup the entry point and perform a
+		 * container connect, therefore this data will be read from root of the new
+		 * container, not the directory.
+		 *
+		 * TODO: This could call getxattr to verify success.
+		 */
+		rc = stat(path, &finfo);
+		if (rc) {
+			rc = errno;
+			D_ERROR("Failed to access new container: %d (%s)\n", rc, strerror(rc));
+			goto err_link;
+		}
+	}
 
 out_cont:
 	rc2 = daos_cont_close(coh, NULL);
-	if (rc == 0)
-		rc = rc2;
+	if (rc2 != -DER_SUCCESS) {
+		DL_ERROR(rc2, "failed to close container");
+		if (rc2 == -DER_NOMEM)
+			// Second close to properly handle fault injection
+			daos_cont_close(coh, NULL);
+		else if (rc == -DER_SUCCESS)
+			rc = daos_der2errno(rc2);
+	}
```

Comment:
I thought we'd designed out these hacks by now but perhaps there's still a few there.  Container close would be a tricky one to re-try at a higher level.